plugins/TTAnalyzer.cc

```diff
@@ -1,118 +1,528 @@
+      m_diLepton.lepIDs[LepLepID::LT] = l1.lepID[LepID::L] && l2.lepID[LepID::T];
+      m_diLepton.lepIDs[LepLepID::TL] = l1.lepID[LepID::T] && l2.lepID[LepID::L];
+      m_diLepton.lepIDs[LepLepID::MM] = l1.lepID[LepID::M] && l2.lepID[LepID::M];
+      m_diLepton.lepIDs[LepLepID::MT] = l1.lepID[LepID::M] && l2.lepID[LepID::T];
+      m_diLepton.lepIDs[LepLepID::TM] = l1.lepID[LepID::T] && l2.lepID[LepID::M];
+      m_diLepton.lepIDs[LepLepID::TT] = l1.lepID[LepID::T] && l2.lepID[LepID::T];
+      m_diLepton.DR = VectorUtil::DeltaR(l1.p4, l2.p4);
+      m_diLepton.DEta = TTAnalysis::DeltaEta(l1.p4, l2.p4);
+      m_diLepton.DPhi = VectorUtil::DeltaPhi(l1.p4, l2.p4);
+
+      diLeptons.push_back(m_diLepton);
     }
+  }
 
-    // El-Mu
-    for(unsigned int iele = 0; iele < selectedElectrons.size(); iele++){
-        for(unsigned int imu = 0; imu < selectedMuons.size(); imu++){
-            if( electrons.p4[ selectedElectrons[iele] ].Pt() > muons.p4[ selectedMuons[imu] ].Pt() && electrons.charge[ selectedElectrons[iele] ] * muons.charge[ selectedMuons[imu] ] < 0 ){
-                selectedLeadingElMu = std::make_pair(iele, imu);
-                break;
-            }
-        }
-        if(selectedLeadingElMu.first >= 0)
-            break;
+  for(uint8_t i = 0; i < diLeptons.size(); i++){
+    const DiLepton& m_diLepton = diLeptons[i];
+    
+    for(const LepLepID::LepLepID& id: LepLepID::it){
+      if(m_diLepton.lepIDs[id])
+        diLeptons_LepIDs[id].push_back(i);
     }
     
-    // Mu-El
-    for(unsigned int imu = 0; imu < selectedMuons.size(); imu++){
-        for(unsigned int iele = 0; iele < selectedElectrons.size(); iele++){
-            if( electrons.p4[ selectedElectrons[iele] ].Pt() < muons.p4[ selectedMuons[imu] ].Pt() && electrons.charge[ selectedElectrons[iele] ] * muons.charge[ selectedMuons[imu] ] < 0 ){
-                selectedLeadingMuEl = std::make_pair(imu, iele);
-                break;
+  }
+
+  ///////////////////////////
+  //       JETS            //
+  ///////////////////////////
+
+  const JetsProducer& jets = producers.get<JetsProducer>("jets");
+
+  // First find the jets passing kinematic cuts and tight jetID
+
+  for(uint8_t ijet = 0; ijet < jets.p4.size(); ijet++){
+      
+    // Save the jets that pass the kinematic cuts
+    if( abs(jets.p4[ijet].Eta()) < m_jetEtaCut && jets.p4[ijet].Pt() > m_jetPtCut){
+      selectedJets.push_back(ijet);
+        
+      // Save the jets that pass the kinematic cuts and tight jetID
+      if( abs(jets.p4[ijet].Eta()) < m_jetEtaCut && jets.p4[ijet].Pt() > m_jetPtCut && jetIDAccessor(jets, ijet, m_jetID)){
+        selectedJets_tightID.push_back(ijet);
+        
+        // Save the jets that pass the kinematic cuts and tight jetID and DR(l,j)>cut using selected leptons, for each DiLepton ID pair
+        for(const LepLepID::LepLepID& id: LepLepID::it){
+          bool passDRCut(true);
+          for(const DiLepton& m_diLepton: diLeptons){
```

Comment:
Any particular reason you iterate over the dileptons pair instead of leptons? Maybe I'm missing something, but I'd find it more natural to have `selectedJets_tightID_DRCut` indexed in lepton ID instead of dilepton-id.